enrichment-web/src/main/java/eu/europeana/enrichment/web/controller/TopicController.java

```diff
@@ -0,0 +1,201 @@
+package eu.europeana.enrichment.web.controller;
+
+import java.util.List;
+
+import java.util.Map;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.EnableCaching;
+import eu.europeana.api.commons.web.http.HttpHeaders;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import eu.europeana.api.commons.web.definitions.WebFields;
+import eu.europeana.api.commons.web.exception.HttpException;
+import eu.europeana.enrichment.exceptions.UnsupportedEntityTypeException;
+import eu.europeana.enrichment.model.TopicEntity;
+import eu.europeana.enrichment.web.exception.ApplicationAuthenticationException;
+import eu.europeana.enrichment.web.model.EnrichmentTopicRequest;
+import eu.europeana.enrichment.web.service.EnrichmentTopicService;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import eu.europeana.enrichment.common.commons.HelperFunctions;
+@RestController
+@EnableCaching
+//@SwaggerSelect
+@Api(tags = "Topic service", description=" ")
+public class TopicController extends BaseRest{
+	
+	@Autowired
+	EnrichmentTopicService enrichmentTopicService;
+	
+	Logger logger = LogManager.getLogger(getClass());
+	
+	
+	
+	/**
+	 * This method represents the /enrichment/topic/ end point,
+	 * where a topic creation request will be processed.
+	 * All requests on this end point are processed here.
+	 * 
+	 * @param wskey
+	 * @return
+	 * @throws Exception 
+	 */
+	@ApiOperation(value = "Create Topic", nickname = "postTopicCreation", notes = "This method stores the topics into the database\n"
+			+ "Mandatory fields: identifier, description and topicTerm")
+	@RequestMapping(value = "/enrichment/topic/", method = {RequestMethod.POST}, produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<String> postTopicCreation (@RequestParam(value = "wskey", required = true) String wskey,
+			@RequestBody(required = false) String text) throws JsonMappingException, JsonProcessingException, HttpException, UnsupportedEntityTypeException
```

Comment:
required true, change name of the param from text to topic and also the type from String to Topic/TopicImpl